pkg/controller/node.go

```diff
@@ -166,25 +168,32 @@ func NewNodeController(mgr manager.Manager, controllerSharedState *SharedState)
 			return true
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			// make sure only update event from node taint changed from "unschedulable" gets enqueued to reconciler queue
+
 			old, oldOk := e.ObjectOld.(*corev1.Node)
 			new, newOk := e.ObjectNew.(*corev1.Node)
 
-			wasUnschedulable := false
-			nowSchedulable := true
-			for _, taint := range old.Spec.Taints {
-				if taint.Key == "node.kubernetes.io/unschedulable" {
-					wasUnschedulable = true
+			if oldOk && newOk {
+				// update event from node taint changed from "unschedulable" gets enqueued to reconciler queue
+				wasUnschedulable := false
+				nowSchedulable := true
+				for _, taint := range old.Spec.Taints {
+					if taint.Key == "node.kubernetes.io/unschedulable" {
+						wasUnschedulable = true
+					}
 				}
-			}
-			for _, taint := range new.Spec.Taints {
-				if taint.Key == "node.kubernetes.io/unschedulable" {
-					nowSchedulable = false
+				for _, taint := range new.Spec.Taints {
+					if taint.Key == "node.kubernetes.io/unschedulable" {
+						nowSchedulable = false
+					}
 				}
-			}
 
-			if oldOk && newOk && wasUnschedulable && nowSchedulable {
-				return true
+				// update event when LabelInstanceTypeStable is added to node
+				_, oldLableOk := old.GetLabels()[v1.LabelInstanceTypeStable]
+				_, newLabelOk := new.GetLabels()[v1.LabelInstanceTypeStable]
+
+				if (wasUnschedulable && nowSchedulable) || (!oldLableOk && newLabelOk) {
```

Comment:
This can be truncated to `return (wasUnschedulable && nowSchedulable) || (!oldLableOk && newLabelOk)`